lib/browser-specific.js

```diff
@@ -0,0 +1,247 @@
+    let smallestIdx = 0;
+    for (let i = 1; i < browserTests.length; i++) {
+      if (browserTests[i].currentKey() < smallestKey) {
+        smallestKey = browserTests[i].currentKey();
+        smallestIdx = i;
+      }
+    }
+    browserTests[smallestIdx].next();
+  }
+
+  // Now recurse into subtrees.
+  const browserSubtrees = browserTrees.map(tree => new IteratorHelper(tree.trees));
+  while (browserSubtrees.every(subtree => subtree.hasNext())) {
+    const firstTree = browserSubtrees[0];
+
+    // If the subtrees are all the same object (which happens due to the caching
+    // in lib/results.js), we can just skip them; it is impossible for there to
+    // be browser-specific failures in the subtree.
+    if (browserSubtrees.every(s => s.currentValue() == firstTree.currentValue())) {
+      browserSubtrees.forEach(s => s.next());
+      continue;
+    }
+
+        if (browserSubtrees.every(s => s.currentKey() == firstTree.currentKey())) {
+      walkTrees(browserSubtrees.map(s => s.currentValue()), scores);
+      browserSubtrees.forEach(s => s.next());
+      continue;
+    }
+
+    // Our iterators are not pointing at the same subtree; find the earliest
+    // iterator and move it forward.
+    let smallestKey = firstTree.currentKey();
+    let smallestIdx = 0;
+    for (let i = 1; i < browserSubtrees.length; i++) {
+      if (browserSubtrees[i].currentKey() < smallestKey) {
+        smallestKey = browserSubtrees[i].currentKey();
+        smallestIdx = i;
+      }
+    }
+    browserSubtrees[smallestIdx].next();
+  }
+}
+
+// Produces a 'score' of browser-specific failures for a given set of runs from
+// different products on the same WPT codebase. The word 'score' is used instead
+// of count as we normalize the counts of subtests.
+//
+// runs: an array of run objects, where each run has the form:
+//       {browser_name: "foo", tree: <an in-memory git tree>}
+//
+// expectedBrowsers: the set of browsers that should be (exactly) represented in
+//                   runs. If a browser is missing, an exception will be thrown.
+//
+// Returns a map from product name to score.
+function scoreBrowserSpecificFailures(runs, expectedBrowsers) {
+  // First, verify that the expected browsers are seen in |runs|.
+  const seenBrowsers = new Set();
+  for (const run of runs) {
+    const browserName = run.browser_name;
+    if (!expectedBrowsers.has(browserName)) {
+      throw new Error(`Unexpected browser found in runs: ${browserName}`);
+    }
+    if (seenBrowsers.has(browserName)) {
+      throw new Error(`${browserName} has multiple entries in runs`);
+    }
+    seenBrowsers.add(browserName);
+  }
+  // Browsers can only be added to seenBrowsers if they were already in
+  // expectedBrowsers (see above), so the only remaining possible error is a
+  // missing browser in the runs.
+  if (seenBrowsers.size != expectedBrowsers.size) {
+    let difference = new Set([...expectedBrowsers].filter(x => !seenBrowsers.has(x)));
```

Comment:
`difference` is immediately spread into an array, so just just let it be an array:

```suggestion
    const difference = [...expectedBrowsers].filter(x => !seenBrowsers.has(x));
```